### PR TITLE
[chore](nereids) Added compatibility with mysql alias conflict

### PR DESCRIPTION
**Pull request by Toms1999**
## Proposed changes

throw table name/alias conflict exception to keep same behavior with mysql

for example:
```sql
select * from test.a b, test.b
```

error:
```
Not unique table/alias: 'b'
```

### CODE REVIEW

**Comment by Toms1999** [comment]
buildall

File: regression-test/suites/nereids_syntax_p0/alias_conflict.groovy [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+// under the License.
+
+suite("alias_conflict") {
+
+    sql """ DROP TABLE IF EXISTS `test1` """
+    sql """ DROP TABLE IF EXISTS `test2` """
+    sql """ DROP TABLE IF EXISTS `test3` """
+
+    sql """
+        CREATE TABLE `test1` (
```
Comment (924060929) [inline regression-test/suites/nereids_syntax_p0/alias_conflict.groovy:25]: `test1` table conflict to `nereids_syntax_p0/test_limit.groovy`, please rename to other table name, e.g. `test_alias_tbl1`, `test_alias_tbl2`

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java [hunk 2]
```diff
@@ -507,6 +508,31 @@ private LogicalJoin<Plan, Plan> bindJoin(MatchingContext<LogicalJoin<Plan, Plan>
...
+                LogicalSubQueryAlias subQueryAlias = (LogicalSubQueryAlias) child;
+                String alias = subQueryAlias.getAlias();
+
+                if (!tableNames.add(alias)) {
+                    throw new AnalysisException("Not unique table/alias: '" + alias + "'");
+                }
+                // get table name
+            } else if (child instanceof LogicalFilter) {
+                Plan grandChild = child.children().get(0);
+                if (grandChild instanceof LogicalOlapScan) {
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java:525]: you should support other table type, for example, `LogicalFileScan` for hive table, so `LogicalRelation` is more suitable

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java [hunk 3]
```diff
@@ -507,6 +508,31 @@ private LogicalJoin<Plan, Plan> bindJoin(MatchingContext<LogicalJoin<Plan, Plan>
         LogicalJoin<Plan, Plan> join = ctx.root;
         CascadesContext cascadesContext = ctx.cascadesContext;
 
+        Set<String> tableNames = new HashSet<>();
+        List<Plan> children = join.children();
+        for (Plan child : children) {
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java:513]: this seems only check one level alias, can you support check conflict names between multiple level?

File: regression-test/suites/nereids_syntax_p0/alias_conflict.groovy [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    // Test for no conflict
+    qt_select_no_conflict """select * from test1 t1, test2 t2 where t1.id = t2.id;"""
+
+    // Complex query with alias conflict
+    test {
+        sql "select * from (select * from test1) a, (select * from test1) a;"
+        exception "Not unique table/alias: 'a'"
+    }
+
```
Comment (924060929) [inline regression-test/suites/nereids_syntax_p0/alias_conflict.groovy:102]: Add more cases

```groovy
multi_sql """
    DROP TABLE IF EXISTS test_alias_tbl1;
    DROP TABLE IF EXISTS test_alias_tbl2;
    CREATE TABLE IF NOT EXISTS `test_alias_tbl1 ` (
      `id` varchar(64) NULL
    )
    DISTRIBUTED BY HASH(`id`)
    PROPERTIES (
      "replication_num"="1"
    );
    CREATE TABLE IF NOT EXISTS `test_alias_tbl2 ` (
      `id` varchar(64) NULL
    )
    DISTRIBUTED BY HASH(`id`)
    PROPERTIES (
      "replication_num"="1"
    );
    """

test {
    sql "select * from test_alias_tbl1, test_alias_tbl1 b, test_alias_tbl1 c, test_alias_tbl1"
    exception "Not unique table/alias: 'test_alias_tbl1'"
}
```

**Comment by 924060929** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java [hunk 5]
```diff
@@ -531,6 +536,33 @@ private LogicalJoin<Plan, Plan> bindJoin(MatchingContext<LogicalJoin<Plan, Plan>
...
+            }
+        } else {
+            // Recursively check the children of the current plan
+            for (Plan child : plan.children()) {
+                checkPlan(child, tableNames);
+            }
+        }
+    }
+
+
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java:565]: how about refactor to
```java
private void checkConflictAlias(Plan plan, Set<String> tableNames, boolean isTopLevel) {
    Set<String> existsTableNames = Sets.newLinkedHashSet();
    Consumer<String> checkAlias = tableAliasName -> {
        if (!existsTableNames.add(tableAliasName)) {
            throw new AnalysisException("Not unique table/alias: '" + tableAliasName + "'");
        }
    };

    boolean stopCheckChildren = true;
    plan.foreach(p -> {
        if (p instanceof LogicalSubQueryAlias) {
            checkAlias.accept(((LogicalSubQueryAlias<?>) p).getAlias());
            return stopCheckChildren;
        } else if (p instanceof LogicalCatalogRelation) {
            TableIf table = ((LogicalCatalogRelation) p).getTable();
            checkAlias.accept(table.getName());
            return stopCheckChildren;
        } else {
            return !stopCheckChildren;
        }
    });
}
```

**Comment by 924060929** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Review by 924060929 — DISMISSED** [review]

File: regression-test/suites/query_p0/cross_db/cross_db.groovy [hunk 6]
```diff
@@ -54,7 +54,7 @@ suite("cross_db") {
         (4, 40)
     """
 
-    sql """SELECT * FROM cross_db_1.cnt_table,cross_db_2.cnt_table"""
+    sql """SELECT * FROM cross_db_1.cnt_table as x,cross_db_2.cnt_table as y"""
```
Comment (morrySnow) [inline regression-test/suites/query_p0/cross_db/cross_db.groovy:57]: why change this case? `SELECT * FROM cross_db_1.cnt_table,cross_db_2.cnt_table` should not throw exception

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java [hunk 7]
```diff
@@ -531,6 +537,29 @@ private LogicalJoin<Plan, Plan> bindJoin(MatchingContext<LogicalJoin<Plan, Plan>
                 join.children(), null);
     }
 
+    private void checkConflictAlias(Plan plan) {
+        Set<String> existsTableNames = Sets.newLinkedHashSet();
+        Consumer<String> checkAlias = tableAliasName -> {
+            if (!existsTableNames.add(tableAliasName)) {
+                throw new AnalysisException("Not unique table/alias: '" + tableAliasName + "'");
+            }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindExpression.java:545]: conflict check should check qualified name

File: regression-test/suites/query_p0/cross_db/cross_db.groovy (hunk 6 above)
Comment (Toms1999) [reply regression-test/suites/query_p0/cross_db/cross_db.groovy:57]: i will fix it

**Comment by Toms1999** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Comment by Toms1999** [comment]
run buildall

**Comment by 924060929** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**